Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/CachedQueries/CoinsConfig.hs

```diff
@@ -19,24 +19,30 @@ import Data.Text (pack)
 import Domain.Types.Coins.CoinsConfig
 import qualified Domain.Types.Merchant as DM
 import qualified Domain.Types.MerchantOperatingCity as DMOC
+import Domain.Types.VehicleCategory as DTV
 import Kernel.Prelude
 import qualified Kernel.Storage.Hedis as Hedis
 import Kernel.Types.Id
 import Kernel.Utils.Common
 import qualified Lib.DriverCoins.Types as DCT
 import qualified Storage.Queries.Coins.CoinsConfig as Queries
 
-fetchFunctionsOnEventbasis :: (CacheFlow m r, EsqDBFlow m r) => DCT.DriverCoinsEventType -> Id DM.Merchant -> Id DMOC.MerchantOperatingCity -> m [CoinsConfig]
-fetchFunctionsOnEventbasis eventType merchantId merchantOpCityId =
-  Hedis.safeGet (makeCoinConfigKey eventType merchantOpCityId) >>= \case
+fetchFunctionsOnEventbasis :: (CacheFlow m r, EsqDBFlow m r) => DCT.DriverCoinsEventType -> Id DM.Merchant -> Id DMOC.MerchantOperatingCity -> DTV.VehicleCategory -> m [CoinsConfig]
+fetchFunctionsOnEventbasis eventType merchantId merchantOpCityId vehicleCategory =
+  Hedis.safeGet (makeCoinConfigKey eventType merchantOpCityId vehicleCategory) >>= \case
     Just a -> return a
-    Nothing -> cacheCoinConfig eventType merchantOpCityId /=<< Queries.fetchFunctionsOnEventbasis eventType merchantId merchantOpCityId
+    Nothing -> cacheCoinConfig eventType merchantOpCityId vehicleCategory /=<< Queries.fetchFunctionsOnEventbasis eventType merchantId merchantOpCityId (Just vehicleCategory)
```

Comment:
What if data based on vehicleCategory is not available e.g. older tables will not have vehicleCategory.